recipes/cpython/all/conanfile.py

```diff
@@ -481,62 +467,80 @@ def _msvc_package_layout(self):
 
         packages = {}
         get_name_version = lambda fn: fn.split(".", 2)[:2]
-        whldir = os.path.join(self._source_subfolder, "Lib", "ensurepip", "_bundled")
+        whldir = os.path.join(self.source_folder, "Lib", "ensurepip", "_bundled")
         for fn in filter(lambda n: n.endswith(".whl"), os.listdir(whldir)):
             name, version = get_name_version(fn)
             add = True
             if name in packages:
                 pname, pversion = get_name_version(packages[name])
-                add = tools.Version(version) > tools.Version(pversion)
+                add = Version(version) > Version(pversion)
             if add:
                 packages[name] = fn
         for fname in packages.values():
-            tools.unzip(filename=os.path.join(whldir, fname), destination=os.path.join(self.package_folder, "bin", "Lib", "site-packages"))
+            unzip(self, filename=os.path.join(whldir, fname),
+                  destination=os.path.join(self.package_folder, "bin", "Lib", "site-packages"))
 
-        self.run("{} -c \"import compileall; compileall.compile_dir('{}')\"".format(os.path.join(build_path, self._cpython_interpreter_name), os.path.join(self.package_folder, self._msvc_install_subprefix, "Lib").replace("\\", "/")),
-                 run_environment=True)
+        interpreter_path = os.path.join(build_path, self._cpython_interpreter_name)
+        lib_dir_path = os.path.join(self.package_folder, self._msvc_install_subprefix, "Lib").replace("\\", "/")
+        self.run(f"{interpreter_path} -c \"import compileall; compileall.compile_dir('{lib_dir_path}')\"")
 
     def package(self):
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
-        if self.settings.compiler == "Visual Studio":
-            if self._is_py2 or not self.options.shared:
-                self._msvc_package_copy()
-            else:
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        if is_msvc(self):
+            if self.options.shared:
                 self._msvc_package_layout()
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "vcruntime*")
+            else:
+                self._msvc_package_copy()
+            rm(self, "vcruntime*", os.path.join(self.package_folder, "bin"), recursive=True)
         else:
-            autotools = self._configure_autotools()
-            autotools.install()
-            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-            tools.rmdir(os.path.join(self.package_folder, "share"))
+            if (os.path.isfile(os.path.join(self.package_folder, "lib"))):
+                # FIXME not sure where this file comes from
+                self.output.info(f"{os.path.join(self.package_folder, 'lib')} exists, but it shouldn't.")
+                rm(self, "lib", self.package_folder)
```

Comment:
```suggestion
            # FIXME not sure where this file comes from
            rm(self, "lib", self.package_folder)
```
Let's simplify: The `rm` only removes files, and does not raise error for file not found.

Reply:
I might remove this hack altogether, I don't know if it ever actually worked. I'll check through some logs to see.

Reply:
It seems intermittent, and actually hasn't happened recently (only checked some recent logs, not going through all of them). I think it's best to just remove it, if it comes up we can just rerun the job.

Reply:
Done, removed the whole condition.